### DISCUSSION TITLE
device code generation question

### DISCUSSION THREAD

**Discussion by jinz2014** [intel/llvm#5615] [Q&A]
Could you please list the commands using clang++/dpcpp to generate LLVM/SPIRV device code ?  It is not clear _how_ and _when_ a user can generate a human-readable device code for debugging and analysis.

Thanks

**Comment by AlexeySachkov** [comment]
Hi @jinz2014,

Please take a look at [SYCL Compiler User Manual](https://github.com/intel/llvm/blob/sycl/sycl/doc/SYCLCompilerUserManual.md#sycl-device-code-compilation)

**Comment by jinz2014** [comment]
It is not clear how users can disassemble a binary to view GPU assembly codes. Could you show some examples ?
